backend/app/api/triggers.py

```
@@ -0,0 +1,185 @@
+    try:
+        query = db.query(Trigger)
+        
+        if workflow_id:
+            query = query.filter(Trigger.workflow_id == workflow_id)
+        
+        triggers = query.all()
+        return [TriggerResponse.from_orm(trigger) for trigger in triggers]
+        
+    except Exception as e:
+        logger.error(f"Error listing triggers: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to list triggers: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)